assets/src/components/detourListPage.tsx

```diff
@@ -10,19 +10,39 @@ import {
   SvgProps,
 } from "../helpers/bsIcons"
 import { DetourModal } from "./detours/detourModal"
-import { fetchDetours } from "../api"
-import { useApiCall } from "../hooks/useApiCall"
-import { isOk } from "../util/result"
 import { joinClasses } from "../helpers/dom"
 import { useLoadDetour } from "../hooks/useLoadDetour"
+import {
+  useActiveDetours,
+  useDraftDetours,
+  usePastDetours,
+} from "../hooks/useDetours"
+import { SocketContext } from "../contexts/socketContext"
+import { SimpleDetour } from "../models/detoursList"
 
 export const DetourListPage = () => {
   const [showDetourModal, setShowDetourModal] = useState(false)
   const [detourId, setDetourId] = useState<number | undefined>()
-  const { result } = useApiCall({
-    apiCall: useCallback(async () => fetchDetours(), []),
-  })
-  const detours = result && isOk(result) && result.ok
+
+  // Wait for the detour channels to initialize
+  const { socket } = useContext(SocketContext)
+
+  const activeDetoursMap = useActiveDetours(socket)
+  const draftDetoursMap = useDraftDetours(socket)
+  const pastDetoursMap = usePastDetours(socket)
+
+  const activeDetours: SimpleDetour[] | null = activeDetoursMap
+    ? Object.values(activeDetoursMap).sort(
+        (a, b) => b.activatedAt - a.activatedAt
+      )
+    : null
+  const draftDetours: SimpleDetour[] | null = draftDetoursMap
+    ? Object.values(draftDetoursMap).sort((a, b) => b.updatedAt - a.updatedAt)
+    : null
+  const pastDetours: SimpleDetour[] | null = pastDetoursMap
+    ? Object.values(pastDetoursMap).sort((a, b) => b.updatedAt - a.updatedAt)
+    : null
```

Comment:
nit(non-blocking): could we simplify these?

```suggestion
  const activeDetours = activeDetoursMap
    && Object.values(activeDetoursMap).sort(
        (a, b) => b.activatedAt - a.activatedAt
      )
  const draftDetours = draftDetoursMap
    && Object.values(draftDetoursMap).sort((a, b) => b.updatedAt - a.updatedAt)
  const pastDetours = pastDetoursMap
    && Object.values(pastDetoursMap).sort((a, b) => b.updatedAt - a.updatedAt)
```

Reply:
Oh of course, ha! Thanks 👍